### PR TITLE
Automatically switch terminal renderer if it's slow

### PR DESCRIPTION
**Pull request by Tyriar**
Part of #53274

### CODE REVIEW

File: src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ import { ansiColorIdentifiers, TERMINAL_BACKGROUND_COLOR, TERMINAL_FOREGROUND_CO
 import { PANEL_BACKGROUND } from 'vs/workbench/common/theme';
 import { IConfigurationService, ConfigurationTarget } from 'vs/platform/configuration/common/configuration';
 import { IEditorOptions } from 'vs/editor/common/config/editorOptions';
-import { INotificationService, Severity, IPromptChoice } from 'vs/platform/notification/common/notification';
```
Comment (dziarkoaka1001) [inline src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts]: 32

**Comment by fessmage** [comment]
How to workaroud situation, when vs code detected slow rendering with renderertype-auto or renderertype=canvas and suggest (or autoswitch) renderertype to dom, but dom render even slower ? I have this situation with my work station. If you apply this PR, i may end up with dom render, which even more slow, then canvas.

**Comment by Tyriar** [comment]
@fessmage I think what you were probably seeing is related to https://github.com/Microsoft/vscode/issues/56628 which was fixed yesterday in [insiders](https://code.visualstudio.com/insiders/).

**Comment by fessmage** [comment]
I was describing behavior ( dom slower than canvas ) from version 1.25.1. Is #56628 related to 1.25.1?

**Comment by Tyriar** [comment]
@fessmage I don't think that slowdown is in 1.26, but then you're describing something from quite an old version...?

**Comment by Tyriar** [comment]
Deferring this until Sept due to https://github.com/Microsoft/vscode/issues/56628 fix going in.

**Comment by Tyriar** [comment]
@fessmage I think we're on the same page, that's why I'm deferring this PR; to see whether it's needed at all anymore.

**Comment by Tyriar** [comment]
Closing for now, as it probably won't be needed after https://github.com/Microsoft/vscode/issues/56628, more improvements also coming in https://github.com/xtermjs/xterm.js/issues/1677

**Pull request closed without merging**